pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -530,78 +585,15 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
                 // The ERC4626 token has a buffer initialized within the Vault. Additionally, since the sender did not
                 // specify an input amount for the wrapped token, the function will wrap the underlying asset and use
                 // the resulting wrapped tokens to add liquidity to the pool.
-                _wrapAndUpdateTokenInAmounts(IERC4626(childToken), params.sender, params.wethIsEth);
+                amountsIn[i] = _wrapAndUpdateTokenInAmounts(IERC4626(childToken), params.sender, params.wethIsEth);
+            } else if (_settledTokenAmounts().tGet(childToken) == 0) {
+                // Set this token's amountIn if it's a standard token that was not previously settled.
+                amountsIn[i] = _currentSwapTokenInAmounts().tGet(childToken);
+                _settledTokenAmounts().tSet(childToken, amountsIn[i]);
```

Comment:
We're only using `_settledTokenAmounts` to check whether a given token appears twice in the tree; is that right?
Shouldn't we revert if that happens? We're not adding amounts anywhere.